pkg/sql/logictest/tests/local/local_test: TestLogic_buffered_writes failed

pkg/sql/logictest/tests/local/local_test.TestLogic_buffered_writes [failed](https://tanzanite.cluster.engflow.com/invocations/default/cfe90fef-b2a3-427c-9423-59c2527f6df7?testReportRun=8&testReportShard=33&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9sb2NhbDpsb2NhbF90ZXN0) on release-25.2 @ [5eb8a34f004be6873a40a1699abeae3fbe640cb5](https://github.com/cockroachdb/cockroach/commits/5eb8a34f004be6873a40a1699abeae3fbe640cb5):


```
=== RUN   TestLogic_buffered_writes
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogic_buffered_writes3358197734
    test_log_scope.go:76: use -show-logs to present logs inline
    logic.go:1570: apply cluster opt logictest.clusterOptDisableUseMVCCRangeTombstonesForPointDeletes
[06:16:02] setting distsql_workmem='32505B';
[06:16:03] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/buffered_writes with config local: 69 tests, 2 failures
--- total: 69 tests, 2 failures
    logic.go:4502: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_buffered_writes3358197734
--- FAIL: TestLogic_buffered_writes (1.95s)
=== RUN   TestLogic_buffered_writes/regression
    logic.go:3111: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/buffered_writes:309: SELECT message FROM [SHOW KV TRACE FOR SESSION] WHERE message LIKE 'CPut %%' OR message LIKE 'Put %%' OR message LIKE 'InitPut %%' OR message LIKE 'Del %%' OR message LIKE 'DelRange %%' OR message LIKE 'ClearRange %%' OR message LIKE 'Get %%' OR message LIKE 'Scan %%' 
        expected:
            Scan /Table/114/2/{2-3}
            Scan /Table/114/1/1/0, /Table/114/1/2/0
            Scan /Table/114/1/2/0
            Put (locking) /Table/114/1/2/0 -> /TUPLE/2:2:Int/3/1:3:Int/3
            Del /Table/114/2/2/2/0
            Put /Table/114/2/3/2/0 -> /BYTES/
            Del /Table/114/3/2/2/0
            Put /Table/114/3/3/2/0 -> /BYTES/
            
        but found (query options: "kvtrace") :
            Scan /Table/114/2/{2-3}
            Scan /Table/114/2/{2/2/0-3}
            Scan /Table/114/1/1/0, /Table/114/1/2/0
            Scan /Table/114/1/2/0
            Put (locking) /Table/114/1/2/0 -> /TUPLE/2:2:Int/3/1:3:Int/3
            Del /Table/114/2/2/2/0
            Put /Table/114/2/3/2/0 -> /BYTES/
            Del /Table/114/3/2/2/0
            Put /Table/114/3/3/2/0 -> /BYTES/
            
    logic.go:2299: 
         /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/buffered_writes:321: too many errors encountered, skipping the rest of the input
--- FAIL: TestLogic_buffered_writes/regression (0.08s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=33</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_buffered_writes.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49528

--- a/pkg/kv/kvclient/kvcoord/txn_interceptor_write_buffer.go
+++ b/pkg/kv/kvclient/kvcoord/txn_interceptor_write_buffer.go
@@ -356,13 +356,13 @@ func (twb *txnWriteBuffer) validateRequests(ba *kvpb.BatchRequest) error {
 			// ReturnRawMVCCValues is unsupported because we don't know how to serve
 			// such reads from the write buffer currently.
 			if t.ReturnRawMVCCValues {
-				return unsupportedOptionError(t.Method(), "ReturnRawMVCCValue")
+				return unsupportedOptionError(t.Method(), "ReturnRawMVCCValues")
 			}
 		case *kvpb.ScanRequest:
 			// ReturnRawMVCCValues is unsupported because we don't know how to serve
 			// such reads from the write buffer currently.
 			if t.ReturnRawMVCCValues {
-				return unsupportedOptionError(t.Method(), "ReturnRawMVCCValue")
+				return unsupportedOptionError(t.Method(), "ReturnRawMVCCValues")
 			}
 			if t.ScanFormat == kvpb.COL_BATCH_RESPONSE {
 				return unsupportedOptionError(t.Method(), "COL_BATCH_RESPONSE scan format")
@@ -371,7 +371,7 @@ func (twb *txnWriteBuffer) validateRequests(ba *kvpb.BatchRequest) error {
 			// ReturnRawMVCCValues is unsupported because we don't know how to serve
 			// such reads from the write buffer currently.
 			if t.ReturnRawMVCCValues {
-				return unsupportedOptionError(t.Method(), "ReturnRawMVCCValue")
+				return unsupportedOptionError(t.Method(), "ReturnRawMVCCValues")
 			}
 			if t.ScanFormat == kvpb.COL_BATCH_RESPONSE {
 				return unsupportedOptionError(t.Method(), "COL_BATCH_RESPONSE scan format")

--- a/pkg/sql/delete_range.go
+++ b/pkg/sql/delete_range.go
@@ -181,10 +181,6 @@ func (d *deleteRangeNode) deleteSpans(params runParams, b *kv.Batch, spans roach
 			if traceKV {
 				log.VEventf(ctx, 2, "DelRange %s - %s", span.Key, span.EndKey)
 			}
-			// TODO(yuzefovich): decide what we do with DeleteRange requests. If
-			// we won't buffer them, then we don't need to make any changes; if
-			// we do buffer them in the interceptor, we'll need to set
-			// to-be-added MustAcquireExclusiveLock flag too.
 			b.DelRange(span.Key, span.EndKey, true /* returnKeys */)
 		}
 	}

--- a/pkg/sql/logictest/testdata/logic_test/buffered_writes
+++ b/pkg/sql/logictest/testdata/logic_test/buffered_writes
@@ -261,64 +261,19 @@ EXPLAIN CREATE DATABASE foo
 statement ok
 EXPLAIN ANALYZE CREATE DATABASE foo
 
-# Regression test for #144273.
+# Regression test for #144273 (not supporting ReturnRawMVCCValues option).
+
+# Get request.
 statement ok
-CREATE TABLE t144273 (
-  k INT PRIMARY KEY,
-  a INT,
-  b INT,
-  INDEX (a),
-  INDEX (b),
-  FAMILY (k, a, b)
-)
+SELECT k, crdb_internal_origin_id FROM t4 WHERE k = 1
 
+# Scan request.
 statement ok
-PREPARE p AS UPDATE t144273 t1 SET a = t2.a + 1, b = t2.b + 1 FROM t144273 t2 WHERE t1.k = t2.a AND t1.k = $1
+SELECT k, crdb_internal_origin_id FROM t4
 
+# ReverseScan request.
 statement ok
-SET vectorize = on
-
-statement ok
-INSERT INTO t144273 VALUES (1, 1, 1);
-
-# The table IDs in the kv trace below are different with the legacy schema
-# changer, so disable that configuration.
-skipif config local-legacy-schema-changer
-query T kvtrace
-EXECUTE p(1)
-----
-Scan /Table/114/2/{1-2}
-Scan /Table/114/1/1/0
-Scan /Table/114/1/1/0
-Put (locking) /Table/114/1/1/0 -> /TUPLE/2:2:Int/2/1:3:Int/2
-Del /Table/114/2/1/1/0
-Put /Table/114/2/2/1/0 -> /BYTES/
-Del /Table/114/3/1/1/0
-Put /Table/114/3/2/1/0 -> /BYTES/
-
-statement ok
-SET vectorize = off
-
-statement ok
-INSERT INTO t144273 VALUES (2, 2, 2);
-
-# The table IDs in the kv trace below are different with the legacy schema
-# changer, so disable that configuration.
-skipif config local-legacy-schema-changer
-query T kvtrace
-EXECUTE p(2)
-----
-Scan /Table/114/2/{2-3}
-Scan /Table/114/1/1/0, /Table/114/1/2/0
-Scan /Table/114/1/2/0
-Put (locking) /Table/114/1/2/0 -> /TUPLE/2:2:Int/3/1:3:Int/3
-Del /Table/114/2/2/2/0
-Put /Table/114/2/3/2/0 -> /BYTES/
-Del /Table/114/3/2/2/0
-Put /Table/114/3/3/2/0 -> /BYTES/
-
-statement ok
-RESET vectorize
+SELECT k, crdb_internal_origin_id FROM t4 ORDER BY k DESC
 
 statement ok
 CREATE TABLE uvw (
@@ -439,6 +394,19 @@ query I
 SELECT count(*) FROM uvw
 ----
 64
+
+# Increase the closed timestamp interval since the transaction below can take
+# long time under race which could result in a RETRY_SERIALIZABLE error on
+# COMMIT (#145065).
+#
+# The cluster setting is SystemVisible, so we must run this query on the host
+# cluster in multi-tenant environment.
+user host-cluster-root
+
+statement ok
+SET CLUSTER SETTING kv.closed_timestamp.target_duration = '30s'
+
+user root
 
 # Create some large blobs that will require the scans to be paginated (in 10MiB
 # chunks).

